x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/nvidia/request/embeddings/NvidiaEmbeddingsRequestTests.java

```diff
@@ -0,0 +1,158 @@
+    private static final InputType INPUT_TYPE_EXPEDIA_IGNORED_VALUE = InputType.SEARCH;
+    private static final Truncation TRUNCATE_EXPEDIA_VALUE = Truncation.START;
+    private static final String INPUT_TYPE_NVIDIA_VALUE = "passage";
+    private static final String INPUT_TYPE_NVIDIA_DEFAULT_VALUE = "query";
+    private static final String TRUNCATE_NVIDIA_VALUE = "start";
+
+    public void testCreateRequest_InputTypeFromTaskSettings_Success() throws IOException {
+        var request = createRequest(URL_VALUE, MODEL_VALUE, INPUT_TYPE_EXPEDIA_INITIAL_VALUE, TRUNCATE_EXPEDIA_VALUE, null);
+        testCreateRequest(request, INPUT_TYPE_NVIDIA_VALUE, TRUNCATE_NVIDIA_VALUE, URL_VALUE);
+    }
+
+    public void testCreateRequest_InputTypeFromRequest_Success() throws IOException {
+        var request = createRequest(URL_VALUE, MODEL_VALUE, null, TRUNCATE_EXPEDIA_VALUE, INPUT_TYPE_EXPEDIA_INITIAL_VALUE);
+        testCreateRequest(request, INPUT_TYPE_NVIDIA_VALUE, TRUNCATE_NVIDIA_VALUE, URL_VALUE);
+    }
+
+    public void testCreateRequest_InputTypeFromRequestPrioritized_Success() throws IOException {
+        var request = createRequest(
+            URL_VALUE,
+            MODEL_VALUE,
+            INPUT_TYPE_EXPEDIA_IGNORED_VALUE,
+            TRUNCATE_EXPEDIA_VALUE,
+            INPUT_TYPE_EXPEDIA_INITIAL_VALUE
+        );
+        testCreateRequest(request, INPUT_TYPE_NVIDIA_VALUE, TRUNCATE_NVIDIA_VALUE, URL_VALUE);
+    }
+
+    public void testCreateRequest_OnlyMandatoryAndDefaultFields_Success() throws IOException {
+        var request = createRequest(null, MODEL_VALUE, null, null, null);
+        testCreateRequest(request, INPUT_TYPE_NVIDIA_DEFAULT_VALUE, null, URL_DEFAULT_VALUE);
+    }
+
+    private void testCreateRequest(NvidiaEmbeddingsRequest request, String expectedInputType, String expectedTruncation, String expectedUrl)
+        throws IOException {
+        var httpRequest = request.createHttpRequest();
+        var httpPost = validateRequestUrlAndContentType(httpRequest, expectedUrl);
+
+        var requestMap = entityAsMap(httpPost.getEntity().getContent());
+        int size = 3;
+        assertThat(requestMap.get(INPUT_FIELD_NAME), Matchers.is(List.of(INPUT_VALUE)));
+        assertThat(requestMap.get(MODEL_FIELD_NAME), Matchers.is(MODEL_VALUE));
+        assertThat(requestMap.get(INPUT_TYPE_FIELD_NAME), Matchers.is(expectedInputType));
+        if (expectedTruncation != null) {
+            size++;
+            assertThat(requestMap.get(TRUNCATE_FIELD_NAME), Matchers.is(expectedTruncation));
+        }
+        assertThat(requestMap, aMapWithSize(size));
+        assertThat(httpPost.getFirstHeader(HttpHeaders.AUTHORIZATION).getValue(), Matchers.is(Strings.format("Bearer %s", API_KEY_VALUE)));
+    }
+
+    public void testCreateRequest_NoModel_ThrowsException() {
+        expectThrows(NullPointerException.class, () -> createRequest(URL_VALUE, null, null, null, null));
+    }
```

Comment:
This test is not actually testing `NvidiaEmbeddingsRequest`; the NPE is bring thrown when constructing the `NvidiaServiceSettings` in `NvidiaEmbeddingsModelTests.createEmbeddingsModel()`, before we can even call the constructor of `NvidiaEmbeddingsRequest`. The test should be removed, or modified to instead pass a null model into the constructor directly.